qiskit/quantum_info/states/stabilizerstate.py

```diff
@@ -214,19 +220,12 @@ def expectation_value(self, oper, qargs=None):
         # Construct Pauli on num_qubits
         pauli = Pauli(num_qubits * "I")
         phase = 0
+        pauli_phase = (-1j) ** oper.phase if oper.phase else 1
 
         for pos, qubit in enumerate(qubits):
-            pauli_pos = (oper.to_label())[len(oper) - 1 - pos]
-            if pauli_pos == "X":
-                pauli.x[qubit] = 1
-            elif pauli_pos == "Y":
-                pauli.x[qubit] = 1
-                pauli.z[qubit] = 1
-                phase += 1
-            elif pauli_pos == "Z":
-                pauli.z[qubit] = 1
-            else:
-                pass
+            pauli.x[qubit] = oper.x[pos]
+            pauli.z[qubit] = oper.z[pos]
+            phase += pauli.x[qubit] & pauli.z[qubit]
```

Comment:
As I understand it, this isn't a behavioural change, right?  It's just a faster way of doing thing.

Reply:
Indeed, this is a faster and correct way, since the previous method did not handle the cases that there was a minus sign, or an i/-i prefix.